packages/core/src/locator.ts

```diff
@@ -1,8 +1,15 @@
 import { ConstructorOrFactory, ClassConstructor } from './internal';
 
 export type Instantiator<Context> = <T>(f: ConstructorOrFactory<Context, T>) => T;
+export type InterceptorGet<Context> = (i: Instantiator<Context>) => Instantiator<Context>;
 
-export type Interceptor<Context> = (i: Instantiator<Context>) => Instantiator<Context>;
+export interface Interceptor<C> {
+  get?: InterceptorGet<C>;
+  has?: <T>(f: ConstructorOrFactory<C, T>) => boolean;
+  override?: <T>(f: ConstructorOrFactory<C, T>, g: ConstructorOrFactory<C, T>) => void;
```

Comment:
Interceptors are used to implement override. They don't have override in them. I think this interface might be too large, just "get" and "inherit" are needed